src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -747,14 +747,14 @@ public void Reset()
             _needsStateRootUpdate = false;
         }
 
-        public void CommitTree()
+        public void CommitTree(long blockNumber)
```

Comment:
not a great fan of this change - maybe worth splitting into CommitTree() and CloseBlock(long blockNumber)?